wrappers/wrapper.rb

```diff
@@ -835,17 +835,285 @@ def check_distances(vrp, unfeasible)
+            vehicle.duration -= rest.duration if vehicle.duration
+            vehicle.timewindow.end -= rest.duration if vehicle.timewindow&.end
+          }
+
+          vehicle[:simplified_rest_ids] = vehicle[:rest_ids].dup
+          vehicle[:rest_ids] = []
+          vehicle[:simplified_rests] = vehicle.rests.dup
+          vehicle.rests = []
+        }
+
+        vrp[:simplified_rests] = vrp.rests.select{ |r| vrp.vehicles.none?{ |v| v.rests.include?(r) } }
+        vrp.rests -= vrp[:simplified_rests]
+      when :rewind
+        # take the modifications back in case the vehicle is moved to another sub-problem
+        vrp.vehicles&.each{ |vehicle|
+          next unless vehicle[:simplified_rest_ids]&.any?
+
+          vehicle[:rest_ids].concat vehicle[:simplified_rest_ids]
+          vehicle[:simplified_rest_ids] = nil
+          vehicle.rests.concat vehicle[:simplified_rests]
+          vehicle[:simplified_rests] = nil
+
+          vehicle.rests.each{ |rest|
+            vehicle.duration += rest.duration if vehicle.duration
+            vehicle.timewindow.end += rest.duration if vehicle.timewindow&.end
+          }
+        }
+
+        if vrp[:simplified_rests]
+          vrp.rests.concat vrp[:simplified_rests]
+          vrp[:simplified_rests] = nil
+        end
+      when :patch_result
+        # correct the result with respect to simplifications
+        vrp.vehicles&.each{ |vehicle|
+          next unless vehicle[:simplified_rest_ids]&.any?
+
+          route = result[:routes].find{ |r| r[:vehicle_id] == vehicle.id }
+          no_cost = route[:activities].none?{ |a| a[:type] == 'service' }
+
+          # first shift every activity all the way to the left (earlier) if the route starts after
+          # the vehicle TW.start so that it is easier to do the insertions since there is no TW on
+          # services, we can do this even if force_start is false
+          shift_amount = vehicle.timewindow&.start.to_i - route[:start_time]
+          shift_route_times(route, shift_amount) if shift_amount < 0
+
+          # insert the rests back into the route and adjust the timing of the activities coming after the pause
+          vehicle[:simplified_rests].each{ |rest|
+            # find the first service that finishes after the TW.end of pause
+            insert_rest_at =
+              unless rest.timewindows&.last&.end.nil?
+                route[:activities].index{ |activity|
+                  (activity[:end_time] || activity[:begin_time]) > rest.timewindows.last.end
+                }
+              end
+
+            insert_rest_at, rest_start =
+              if insert_rest_at.nil?
+                # reached the end of the route or there is no TW.end on the pause
+                # in any case, insert the rest at the end (before the end depot if it exists)
+                if route[:activities].empty?
+                  # no activity
+                  [route[:activities].size, vehicle.timewindow&.start || 0]
```

Comment:
Timewindow start is 0 by default.

```suggestion
                  [route[:activities].size, vehicle.timewindow&.start]
```

Reply:
Do we always have a TW by default though?

Reply:
No we don't :)